{
	"documenttitle": {
		"text": "Red Hat Security Advisory: python38:3.8 and python38-devel:3.8 security update",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHSA-2023:2763"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2023-05-16T08:43:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2023-05-16T08:43:00Z",
		"CurrentReleaseDate": "2023-05-16T08:43:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.14.1",
			"Date": "2023-05-16T14:17:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nAn update for the python38:3.8 and python38-devel:3.8 modules is now available for Red Hat Enterprise Linux 8.\n\nRed Hat Product Security has rated this update as having a security impact of Moderate. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE link(s) in the References section.\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\nPython is an interpreted, interactive, object-oriented programming language, which includes modules, classes, exceptions, very high level dynamic data types and dynamic typing. Python supports interfaces to many system calls and libraries, as well as to various windowing systems. \n\nThe following packages have been upgraded to a later upstream version: python38 (3.8.16).\n\nSecurity Fix(es):\n\n* python: int() type in PyLong_FromString() does not limit amount of digits converting text to int leading to DoS (CVE-2020-10735)\n\n* python: open redirection vulnerability in lib/http/server.py may lead to information disclosure (CVE-2021-28861)\n\n* python: CPU denial of service via inefficient IDNA decoder (CVE-2022-45061)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n\nAdditional Changes:\n\nFor detailed information on changes in this release, see the Red Hat Enterprise Linux 8.8 Release Notes linked from the References section.\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2023 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Moderate",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHSA-2023:2763",
				"Description": "https://access.redhat.com/errata/RHSA-2023:2763"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/security/updates/classification/#moderate",
				"Description": "https://access.redhat.com/security/updates/classification/#moderate"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.8_release_notes/index",
				"Description": "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.8_release_notes/index"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "A flaw was found in python. In algorithms with quadratic time complexity using non-binary bases, when using int(\"text\"), a system could take 50ms to parse an int string with 100,000 digits and 5s for 1,000,000 digits (float, decimal, int.from_bytes(), and int() for binary bases 2, 4, 8, 16, and 32 are not affected). The highest threat from this vulnerability is to system availability.",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2022-09-02T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2020-10735",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2023:2763"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://www.cve.org/CVERecord?id=CVE-2020-10735",
						"Description": "https://www.cve.org/CVERecord?id=CVE-2020-10735"
					},
					{
						"URL": "https://nvd.nist.gov/vuln/detail/CVE-2020-10735",
						"Description": "https://nvd.nist.gov/vuln/detail/CVE-2020-10735"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2020-10735",
						"Description": "CVE-2020-10735"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1834423",
						"Description": "bz#1834423: int() type in PyLong_FromString() does not limit amount of digits converting text to int leading to DoS"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2020-05-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.8.0.GA:python38:3.8-8080020221221151857.0d9ba776",
						"CRB-8.8.0.GA:python38-devel:3.8-8080020221221151857.0d9ba776"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "2",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "A vulnerability was found in python. This security flaw causes an open redirection vulnerability in lib/http/server.py due to no protection against multiple (/) at the beginning of the URI path. This issue may lead to information disclosure.",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2022-08-22T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-28861",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2023:2763"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://www.cve.org/CVERecord?id=CVE-2021-28861",
						"Description": "https://www.cve.org/CVERecord?id=CVE-2021-28861"
					},
					{
						"URL": "https://nvd.nist.gov/vuln/detail/CVE-2021-28861",
						"Description": "https://nvd.nist.gov/vuln/detail/CVE-2021-28861"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-28861",
						"Description": "CVE-2021-28861"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=2120642",
						"Description": "bz#2120642: open redirection vulnerability in lib/http/server.py may lead to information disclosure"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2022-08-23T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.8.0.GA:python38:3.8-8080020221221151857.0d9ba776",
						"CRB-8.8.0.GA:python38-devel:3.8-8080020221221151857.0d9ba776"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "3",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "A vulnerability was discovered in Python. A quadratic algorithm exists when processing inputs to the IDNA (RFC 3490) decoder, such that a crafted unreasonably long name being presented to the decoder could lead to a CPU denial of service. Hostnames are often supplied by remote servers that could be controlled by a malicious actor, which could trigger excessive CPU consumption on the client attempting to make use of an attacker-supplied hostname.",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2022-11-09T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2022-45061",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2023:2763"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://www.cve.org/CVERecord?id=CVE-2022-45061",
						"Description": "https://www.cve.org/CVERecord?id=CVE-2022-45061"
					},
					{
						"URL": "https://nvd.nist.gov/vuln/detail/CVE-2022-45061",
						"Description": "https://nvd.nist.gov/vuln/detail/CVE-2022-45061"
					},
					{
						"URL": "https://github.com/python/cpython/issues/98433",
						"Description": "https://github.com/python/cpython/issues/98433"
					},
					{
						"URL": "https://python-security.readthedocs.io/vuln/slow-idna-large-strings.html",
						"Description": "https://python-security.readthedocs.io/vuln/slow-idna-large-strings.html"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2022-45061",
						"Description": "CVE-2022-45061"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=2144072",
						"Description": "bz#2144072: CPU denial of service via inefficient IDNA decoder"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2022-11-18T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"AppStream-8.8.0.GA:python38:3.8-8080020221221151857.0d9ba776",
						"CRB-8.8.0.GA:python38-devel:3.8-8080020221221151857.0d9ba776"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		}
	],
	"producttree": {
		"xmlns": "http://www.icasi.org/CVRF/schema/prod/1.1",
		"branch": [
			{
				"type": "Product Family",
				"name": "Red Hat Enterprise Linux",
				"branch": [
					{
						"type": "Product Name",
						"name": "Red Hat Enterprise Linux AppStream (v. 8)",
						"fullproductname": {
							"text": "Red Hat Enterprise Linux AppStream (v. 8)",
							"productid": "AppStream-8.8.0.GA",
							"cpe": "cpe:/a:redhat:enterprise_linux:8::appstream"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Enterprise Linux CRB (v. 8)",
						"fullproductname": {
							"text": "Red Hat Enterprise Linux CRB (v. 8)",
							"productid": "CRB-8.8.0.GA",
							"cpe": "cpe:/a:redhat:enterprise_linux:8::crb"
						}
					}
				],
				"fullproductname": {}
			},
			{
				"type": "Product Version",
				"name": "python38-devel:3.8-8080020221221151857.0d9ba776",
				"fullproductname": {
					"text": "python38-devel-3.8-8080020221221151857.0d9ba776",
					"productid": "python38-devel:3.8-8080020221221151857.0d9ba776",
					"cpe": "cpe:/a:redhat:enterprise_linux:8::crb"
				}
			},
			{
				"type": "Product Version",
				"name": "python38:3.8-8080020221221151857.0d9ba776",
				"fullproductname": {
					"text": "python38-3.8-8080020221221151857.0d9ba776",
					"productid": "python38:3.8-8080020221221151857.0d9ba776",
					"cpe": "cpe:/a:redhat:enterprise_linux:8::appstream"
				}
			}
		],
		"relationship": [
			{
				"productreference": "python38:3.8-8080020221221151857.0d9ba776",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "AppStream-8.8.0.GA",
				"fullproductname": {
					"text": "python38:3.8-8080020221221151857.0d9ba776 as a component of Red Hat Enterprise Linux AppStream (v. 8)",
					"productid": "AppStream-8.8.0.GA:python38:3.8-8080020221221151857.0d9ba776",
					"cpe": "cpe:/a:redhat:enterprise_linux:8::appstream"
				}
			},
			{
				"productreference": "python38-devel:3.8-8080020221221151857.0d9ba776",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "CRB-8.8.0.GA",
				"fullproductname": {
					"text": "python38-devel:3.8-8080020221221151857.0d9ba776 as a component of Red Hat Enterprise Linux CRB (v. 8)",
					"productid": "CRB-8.8.0.GA:python38-devel:3.8-8080020221221151857.0d9ba776",
					"cpe": "cpe:/a:redhat:enterprise_linux:8::crb"
				}
			}
		]
	}
}
